lib/apiForSheets.js

```diff
@@ -32,65 +30,72 @@ module.exports.authorize = function(credentials, callback) {
       }
     }
 
-    if (claFlag === true) {
-      for (var row in rows) {
-        var rowUserName = rows[row][0];
-        if (rowUserName === userName) {
-          hasUserSignedCla = true;
-          break;
-        }
+    // Check if the author has signed the CLA.
+    for (var row in rows) {
+      var rowUserName = rows[row][0];
+      if (rowUserName === userName) {
+        hasUserSignedCla = true;
+        break;
       }
+    }
+
+    // If the PR has a CLA label, check if the author has signed the CLA.
+    if (claFlag === true) {
+      // If author has signed the CLA, remove the CLA label.
       if (hasUserSignedCla === true) {
-        await context.github.issues.removeLabel(context.issue({
+        await context.github.issues.removeLabel(context.repo({
+          number: prNumber,
           name: claLabel[0]
         }));
       }
       return;
     }
 
-    if (isPullRequest === true) {
-      for (var row in rows) {
-        var rowUserName = rows[row][0];
-        if (rowUserName === userName) {
-          hasUserSignedCla = true;
-          break;
-        }
-      }
-      if (hasUserSignedCla === false) {
-        var linkText = 'here';
-        var linkResult = linkText.link(
-          'https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#setting-things-up');
-        params = context.issue({
-          body: 'Hi! @' + userName +
-              '. Welcome to Oppia! Please could you ' +
-              'follow the instructions ' + linkResult +
-              ' to get started? You\'ll need to do ' +
-              'this before we can accept your PR. Thanks!'});
-        await context.github.issues.addLabels(context.issue({
-          labels: claLabel
-        }));
-        await context.github.issues.createComment(params);
-        commentCreatedStatus = true;
-        return commentCreatedStatus;
-      }
+    // If the PR does not have a CLA label and the author has signed
+    // the CLA, no action is required.
+    if (hasUserSignedCla === true) {
+      return;
+    }
+
+    // If the PR does not have a CLA label and the author has not signed
+    // the CLA, ping the author in the PR thread to sign the CLA.
+    if (claFlag === false && hasUserSignedCla === false) {
+      var linkText = 'here';
+      var linkResult = linkText.link(
+        'https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#setting-things-up');
+      var params = context.repo({
+        number: prNumber,
+        body: 'Hi! @' + userName +
+            '. Welcome to Oppia! Please could you ' +
+            'follow the instructions ' + linkResult +
+            ' to get started? You\'ll need to do ' +
+            'this before we can accept your PR. Thanks!'});
+      labelPromiseObj = await context.github.issues.addLabels(context.repo({
+        number: prNumber,
+        labels: claLabel}));
+      await context.github.issues.createComment(params);
+      commentCreatedStatus = true;
+      return commentCreatedStatus;
```

Comment:
Just wondering, why do we return true here when we don't return anything anywhere else?

Reply:
We use this in the spec to check the comment creation status: https://github.com/oppia/oppiabot/blob/master/spec/oppiabotSpec.js#L170